src/Translator.zig

```diff
@@ -592,9 +592,55 @@ fn transFnDecl(t: *Translator, fn_decl_node: Node.Index, is_pub: bool) Error!voi
     return t.addTopLevelDecl(fn_name, proto_node);
 }
 
-fn transVarDecl(t: *Translator, variable: Node.Variable) Error!void {
+fn transVarDecl(t: *Translator, scope: *Scope, variable: Node.Variable) Error!void {
     const name = t.tree.tokSlice(variable.name_tok);
-    return t.failDecl(variable.name_tok, name, "TODO translate variable declaration", .{});
+    const toplevel = scope.id == .root;
+    const bs: *Scope.Block = if (!toplevel) try scope.findBlockScope(t) else undefined;
+
+    const type_node = t.transType(scope, variable.qt, variable.name_tok) catch |err| switch (err) {
+        error.UnsupportedType => {
+            return t.failDecl(variable.name_tok, name, "unable to translate variable declaration type", .{});
+        },
+        else => |e| return e,
+    };
+
+    const init_node = if (variable.initializer) |init|
+        t.transExprCoercing(scope, init, .used) catch |err| switch (err) {
+            error.UnsupportedTranslation, error.UnsupportedType => {
+                return t.failDecl(variable.name_tok, name, "unable to resolve var init expr", .{});
+            },
+            else => |e| return e,
+        }
+    else
+        ZigTag.undefined_literal.init();
+
+    const alignment: ?c_uint = variable.qt.requestedAlignment(t.comp) orelse null;
+    const payload = try t.arena.create(ast.Payload.VarDecl);
+    payload.* = .{
+        .base = .{ .tag = ZigTag.var_decl },
+        .data = .{
+            .is_pub = false,
```

Comment:
```suggestion
            .is_pub = toplevel,
```
There may be an option to make `static` variables and functions not pub in the future but for now everything should be pub by default.